app/services/step_by_step_filter.rb

```diff
@@ -1,11 +1,25 @@
 class StepByStepFilter
-  attr_reader :status
+  attr_reader :status, :title_or_url
```

Comment:
All the example test cases in this (second) commit test against the title, but not the URL. I think it would be cleaner to have `:title` only in this commit, and expand it out to be `:title_or_url` in a future commit. (That, or leave the code as-is but write additional tests against the URL in this commit)

~I'd also suggest renaming URL to `slug`, it's a little more specific and is the field that is actually getting matched against.~ EDIT: now I've gone through the other commits, I can see it matches on full URLs as well as slugs.

Reply:
I thought about calling it `title` and then renaming it, but thought it was unnecessary as the filtering by slug/base_path/url would start to be built in the next commit, and renaming variables is extra noise when looking at changes. I'd rather keep the URL tests separate and they need specific changes to the search query to work.